internal/telemetry/segment.go

```diff
@@ -150,7 +149,7 @@ func (s *SegmentClient) send(ctx context.Context, es EventState, et EventType, e
 		return fmt.Errorf("unable to create request body: %w", err)
 	}
 
-	req, err := http.NewRequestWithContext(ctx, http.MethodPost, url, bytes.NewBuffer(data))
+	req, err := http.NewRequest(http.MethodPost, url, bytes.NewBuffer(data))
```

Comment:
What if we only made this change, but still passed the context through everything?

Reply:
I'm thinking about this in a similar way that the go slog package handles cancelled contexts, in that they ignore them in certain cases; e.g.
> The signature of Handle is
>  Handle(context.Context, Record) error
>
> The context is provided to support applications that provide logging information along the call chain. In a break with usual Go practice, the Handle method should not treat a canceled context as a signal to stop work.

Reply:
Sure, that's a good idea.